test/src/test/java/org/corfudb/integration/CorfuStoreIT.java

```diff
@@ -232,4 +244,122 @@ public void readDataWithDynamicMessages() throws Exception {
 
         assertThat(shutdownCorfuServer(corfuServer)).isTrue();
     }
+
+    public Token checkpointAndTrimCorfuStore(CorfuRuntime runtimeC, boolean skipTrim, String tempDiskPath) {
+
+        TableRegistry tableRegistry = runtimeC.getTableRegistry();
+        CorfuTable<CorfuStoreMetadata.TableName,
+                CorfuRecord<CorfuStoreMetadata.TableDescriptors,
+                        CorfuStoreMetadata.TableMetadata>>
+                        tableRegistryCT = tableRegistry.getRegistryTable();
+
+        // Save the regular serializer first..
+        ISerializer protobufSerializer = Serializers.getSerializer(ProtobufSerializer.PROTOBUF_SERIALIZER_CODE);
+
+        // Must register dynamicProtobufSerializer *AFTER* the getTableRegistry() call to ensure that
+        // the serializer does not go back to the regular ProtobufSerializer
+        ISerializer dynamicProtobufSerializer = new DynamicProtobufSerializer(runtimeC);
+        Serializers.registerSerializer(dynamicProtobufSerializer);
+
+        // First checkpoint the TableRegistry system table
+        MultiCheckpointWriter<CorfuTable> mcw = new MultiCheckpointWriter<>();
+
+        Token trimToken = new Token(Token.UNINITIALIZED.getEpoch(), Token.UNINITIALIZED.getSequence());
+        for (CorfuStoreMetadata.TableName tableName : tableRegistry.listTables(null)) {
+            String fullTableName = TableRegistry.getFullyQualifiedTableName(
+                    tableName.getNamespace(), tableName.getTableName()
+                    );
+            SMRObject.Builder<CorfuTable<CorfuDynamicKey, CorfuDynamicRecord>> corfuTableBuilder = runtimeC.getObjectsView().build()
+                    .setTypeToken(new TypeToken<CorfuTable<CorfuDynamicKey, CorfuDynamicRecord>>() {})
+                    .setStreamName(fullTableName)
+                    .setSerializer(dynamicProtobufSerializer);
+
+            // Find out if a table needs to be backed up by disk path to even checkpoint
+            boolean diskBased = tableRegistryCT.get(tableName).getMetadata().getDiskBased();
+            if (diskBased) {
+                final Path persistedCacheLocation = Paths.get(tempDiskPath + tableName.getTableName());
+                final Options options = new Options().setCreateIfMissing(true);
+                final Supplier<StreamingMap<CorfuDynamicKey, CorfuDynamicRecord>> mapSupplier = () -> new PersistedStreamingMap<>(
+                        persistedCacheLocation, options,
+                        dynamicProtobufSerializer, runtimeC);
+                corfuTableBuilder.setArguments(mapSupplier, ICorfuVersionPolicy.MONOTONIC);
+            }
+
```

Comment:
I know you are just using the provided API, but we really need to clean this interface. Creating a disk back CorfuTable should be as easy as creating an in-memory table.

Reply:
I completely agree. The api is super messy right now.